src/javascript/JContent/ContentNavigation/ContentNavigation.container.jsx

```diff
@@ -1,47 +1,68 @@
-import React from 'react';
-import {connect, useSelector} from 'react-redux';
-import {cmGoto} from '../JContent.redux';
+import React, {useMemo} from 'react';
+import {useSelector, useDispatch} from 'react-redux';
 import ContentNavigation from './ContentNavigation';
 import PropTypes from 'prop-types';
 import {registry} from '@jahia/ui-extender';
 import {useNodeChecks} from '@jahia/data-helper';
 import JContentConstants from '../JContent.constants';
+import {createStructuredSelector} from 'reselect';
+import {cmGoto} from '~/JContent/JContent.redux';
+import NavigationHeader from '~/JContent/ContentNavigation/NavigationHeader';
 
-const ContentNavigationContainer = ({mode, siteKey, handleNavigation}) => {
-    let accordionItems = registry.find({type: 'accordionItem', target: 'jcontent'});
-
-    const currentState = useSelector(state => ({site: state.site, language: state.language}));
+const ContentNavigationContainer = ({handleNavigation, selectorObject, accordionItemTarget, accordionItemType, header, sitePermissions}) => {
+    const selector = useMemo(() => createStructuredSelector(selectorObject), [selectorObject]);
+    const dispatch = useDispatch();
+    const {siteKey, language, mode} = useSelector(state => selector(state));
```

Comment:
I don't understand the usage of reselect here and the format of the selectorObject - why not directly passing the selector function (arg of useSelector) as a prop ? default value would be `state => ({site: state.site, language: state.language, mode: state.jcontent.mode})` , and content editor picker could pass `state => ({site: state.contenteditor.picker.site,lang: state.language, path: state.contenteditor.picker.path,openPaths: state.contenteditor.picker.openPaths})`

The memoizations that reselect brings is not of any value here as we just read some properties into the state, no heavy calculation is involved. It will actually memoize the 3 individual selector (site, language, mode) and the structured one for no benefit.

Reply:
Not a fan of hook overload myself, I used reselect primarily because they claim to return a memoized object which prevents rerendering on every state change which happens if we simply supply an object or function supplying an object. I don't know how significant the change is in our context but if we are ok with some extra rerendering I can get rid of reselect.

Reply:
I think I get your point - useSelector will probably trigger a re-render on every redux state change - and the reselect should skip these rerender with the memoization. However this is a problem we have on a lot of useSelector usage (when we return an object), and maybe we should address that more globally in a dedicated ticket. We would need to check the balance between avoiding render and adding memo. 
Also a more simple solution (without memo), when we are just getting {siteKey,language,mode}, would be to pass an equalityFn to useSelector ( from https://react-redux.js.org/api/hooks#equality-comparisons-and-updates , `const selectedData = useSelector(selectorReturningObject, shallowEqual)`)

Reply:
Ok, sounds good, then I will remove reselect :)